ui/pages/settings/add/access-key.js

```diff
@@ -0,0 +1,345 @@
+    }
+
+    return false
+  }
+
+  return (
+    <div className='mx-auto w-full max-w-2xl'>
+      <Head>
+        <title>Add Access Key - Infra</title>
+      </Head>
+      <div className='flex items-center justify-between'>
+        <h1 className='my-6 py-1 font-display text-xl font-medium'>
+          {`New access key`}
+        </h1>
+        <Link href='/settings'>
+          <XMarkIcon
+            className='h-5 w-5 text-gray-500 hover:text-gray-800'
+            aria-hidden='true'
+          />
+        </Link>
+      </div>
+
+      <div className='flex w-full flex-col'>
+        <form onSubmit={onSubmit} className='mb-6 flex flex-col space-y-6'>
+          <div className='flex flex-col space-y-1'>
+            <label className='text-2xs font-medium text-gray-700'>Name</label>
+            <input
+              name='name'
+              required
+              autoFocus
+              spellCheck='false'
+              type='search'
+              onKeyDown={e => {
+                if (e.key === 'Escape' || e.key === 'Esc') {
+                  e.preventDefault()
+                }
+              }}
+              value={name}
+              onChange={e => {
+                setName(e.target.value)
+                setError('')
+              }}
+              className={`mt-1 block w-full rounded-md shadow-sm focus:border-blue-500 focus:ring-blue-500 sm:text-sm ${
+                error ? 'border-red-500' : 'border-gray-300'
+              }`}
+            />
+          </div>
+          <div className='flex flex-col space-y-1'>
+            <label className='text-2xs font-medium text-gray-700'>
+              Expiration
+            </label>
+            <div className='flex flex-col sm:flex-row sm:items-center'>
+              <ExpirationRateMenu
+                selected={selectedTTL}
+                setSelected={setSelectedTTL}
+              />
+            </div>
+          </div>
+          {selectedTTL?.value && (
+            <div className='space-y-1 pt-6 text-xs text-gray-500'>
+              {selectedTTL.default ? (
+                <div>
+                  This access key will expire on{' '}
+                  <span className='font-semibold text-gray-900'>
+                    {moment()
+                      .add(
+                        parseInt(selectedTTL.value),
+                        selectedTTL.value.charAt(selectedTTL.value.length - 1)
+                      )
+                      .format('h:mm:ss a, MMMM Do YYYY')}
+                  </span>
+                </div>
+              ) : (
+                <>
+                  <div>
+                    This access key must be used at least once every{' '}
+                    <span className='font-semibold text-gray-900'>30 days</span>
+                    ,
+                  </div>
+                  <div>
+                    and will expire on{' '}
+                    <span className='font-semibold text-gray-900'>
+                      {moment()
```

Comment:
It feels a bit odd to me that this updates every time I type in the name and the page re-renders. Should we just be more vague and give the date without the time?

Ex: `February 13th, 2023` rather than `10:54:58 am, February 13th 2023`

We can display a more accurate expiry after creation if we want to be exact

Reply:
Patrick suggested to add the time. What do you think @pdevine